### DISCUSSION TITLE
QUERY - Papderdoll system of Black Gate vs Serpent Isle

### DISCUSSION THREAD

**Discussion by Tygaran** [exult/exult#412] [General]
Exult let's you activate an SI style paper-doll for use while playing the Black Gate.  Cool enough, but it's not exactly the same.  The main standouts seem to be th ability to wear both a cloak AND a necklace in SI, and the abikity to wear two rings along with gauntlets in SI.  In BG, there seems to be only one 'neck' slot and only one 'hands' slot.

For modding based on the Black Gate, is there any way to update the paper doll-ing to make full use of the SI equipment slots?

**Comment by DominusExult** [comment]
The problem is that you can't have more slots than the original BG paperdoll provided. For the simple reason that you can toggle the original and SI style paperdoll. So items in the + slots would vanish when you toggle the original paperdoll.

**Comment by Tygaran** [comment]
Crazy followup question then, it it feasible to make a Serpent Isle mod out of the Black Gate data?  Essentially play BG under the SI sub-engine?

I'm basically looking for a quick way to start working with the Britainnia map with the SI methods and assets

**Comment by wench** [comment]
You could do that, There wouldn't be anything in particular that would prevent it, except for the fact that the intro and endgames sequences are hard coded to each specific game. So you'd need to have a modified exult that fixes that and if you are going to be modifying Exult you might as well just go in and change the Behaviour of BG and make it act like SI. It would certainly be easier than porting all the BG usecode to SI engine behaviour.

**Comment by Tygaran** [comment]
Here's a different question, but related to the main topis so I'm not starting a new thread.

I'm trying to get a handle on the file structures of exult as they relate to paperdolls.  Mainly, I'm wondering how Black Gate 'inherits' the main paperdoll images.  If my old notes are correct, exult_bg.flx contatins a sub-file which provides all the paperdoll image unique to BG (3.u7o?), but where is Exult getting the images its 'inheriting' from Serpent Isle?

**Comment by wench** [comment]
when playing Black Gate Exult will attempt to load serpent isle paper doll graphics from your configured installation of Serpent Isle and only from there. At startup Exult checks what games have been configured and Exults sets things up so any game can load data from an SI install. It is something of a hack I added but its useful enough that its stayed in the code for more than 20 years.

**Comment by jharrison03** [comment]
Am I the only person that finds the paper dolls harder to actually use?  Yes, they look better, but it's difficult to manipulate some items like rings or ammunition for ranged weapons.

**Comment by DominusExult** [comment]
> Am I the only person that finds the paper dolls harder to actually use? Yes, they look better, but it's difficult to manipulate some items like rings or ammunition for ranged weapons.

Yes, you are the only one :)
@wench did great stuff with Exult but that hack is an iconic addition :)
But yes, some things are bit more cumbersome but worth it, IMO.

**Comment by jharrison03** [comment]
Spark is bad enough.  How does U6 project handle Sherry?

**Reply by GrantWChapman** [reply]
I haven't played far enough to see how they handle her, but I can confirm that Sherry doesn't have paperdoll art in the U6 Remake by Zygon. I just cheated her into my party and she uses the caucasian female avatar paperdoll art.
She doesn't have Black Gate paperdoll art either when the SI art is turned off.

**Comment by jharrison03** [comment]
![Untitled](https://github.com/exult/exult/assets/44394516/bd44c9c1-12df-40bd-9538-f2bcc5a3c5d4)

**Reply by KnightCaptainU7** [reply]
There is a unique BG doll card for an emp, under gumps.vga it is shape 23:
![image](https://github.com/exult/exult/assets/117867844/ccb8fc9e-1c50-4c34-b43b-df9dbef87166)

You could try setting in on this emp shape:
![image](https://github.com/exult/exult/assets/117867844/3a20a469-e5b2-471d-89ea-4143fef91d0c)

**Comment by KnightCaptainU7** [comment]
It would be awesome if there was a way to take the art from the two games (Forge + Silver Seed) and merge them somehow under the SI code base. Sadly the work to do that would be substantial to make it run on user systems (to avoid distributing the end result directly).
